### PR TITLE
Expose object store(s) availability as a readiness check

### LINKED ISSUE #8833: [Catalog] Check object store connections
We should have a way to check all object store configurations.

Options:
* during startup - this may block startup for too long, OTOH ensures that object stores are already initialized
* using an administrative endpoint - as a health/readiness check

(Not fully thought through - but something like this should help / improve UX)

### PR DESCRIPTION
**Pull request by snazy**
Sample of a successful check:
```json
{ "status": "UP",
  "checks": [
    {
      "name": "Warehouses Object Stores",
      "status": "UP",
      "data": {
        "warehouse.warehouse.object-store.availability": true
      }
} ] }
```

Sample of a failed check (with a wrong S3 access key):
```json
{ "status": "DOWN",
  "checks": [ {
    "name": "Warehouses Object Stores",
    "status": "DOWN",
    "data": {
      "warehouse.warehouse.object-store.availability": false,
      "warehouse.warehouse.object-store.error": "java.io.IOException: software.amazon.awssdk.services.s3.model.S3Exception: The Access Key Id you provided does not exist in our records. (Service: S3, Status Code: 403, Request ID: 17DB0B619B94C358, Extended Request ID: dd9025bab4ad464b049177c95eb6ebf374d3b3fd1af9251148b658df7ac2e3e8)"
  } } ] }
```

Fixes #8833

### CODE REVIEW

File: catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3ObjectIO.java [hunk 1]
```diff
@@ -45,6 +45,16 @@ public S3ObjectIO(S3ClientSupplier s3clientSupplier, Clock clock) {
     this.clock = clock;
   }
 
+  @Override
+  public void ping(StorageUri uri) throws IOException {
+    S3Client s3client = s3clientSupplier.getClient(uri);
+    try {
+      s3client.getBucketLocation(b -> b.bucket(uri.requiredAuthority()));
```
Comment (dimas-b) [inline catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3ObjectIO.java:52]: This is fine from my POV, but it probably requires extra permissions in AWS.

**Review by dimas-b — APPROVED** [review]

File: catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3ObjectIO.java (hunk 1 above)
Comment (snazy) [reply catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3ObjectIO.java:52]: For the bucket-location? Uh

Comment (dimas-b) [reply catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3ObjectIO.java:52]: Also `S3Client` javadoc has: `This operation is not supported by directory buckets.`

Comment (snazy) [reply catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3ObjectIO.java:52]: Oh! there's a `headBucket` anyways

Comment (snazy) [reply catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3ObjectIO.java]: rebased + updated this one

Comment (dimas-b) [reply catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3ObjectIO.java]: +1 to `headBucket`. MinIO does not appear to support `GetBucketLocaltion`: https://min.io/docs/minio/linux/reference/s3-api-compatibility.html

**Review by dimas-b — APPROVED** [review]

**Pull request merged**